test.py

```diff
@@ -222,6 +222,60 @@ def inference_args(parser, remaining_args):
     return inf_parser.parse_args()
 
 
+def test_main(input_files, samplerate, niter, alpha, softmask, residual_model, model,
+              targets=('vocals', 'drums', 'bass', 'other'), outdir=None, no_cuda=False):
```

Comment:
* `targets` should be a list for consistency
* since now this is becomes another interface to the separate function, the defaults arguments should be set for all parameters so that they match the argparse defaults
* I would prefer not-hanging identation to match the rest of the code

Reply:
> * `targets` should be a list for consistency

I made it a tuple so it's immutable since mutable parameters actually give an error and should not be used. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments
> * since now this is becomes another interface to the separate function, the defaults arguments should be set for all parameters so that they match the argparse defaults

Okay, done. I hadn't done it earlier since it was redundant because the separate function already had default parameters for the ones I had left out:
`def separate(audio, targets, model_name='umxhq', niter=1, softmask=False, alpha=1.0, residual_model=False, device='cpu'):`
> * I would prefer not-hanging identation to match the rest of the code

I'm not sure what you mean, I followed the PEP8 guidelines of limiting a line to 120 chars, do you want me to do it line by line? (I think that might make it too long)

Reply:
Also congratulations on getting it published!